src/lsp/lp_doc.ml

```diff
@@ -12,6 +12,106 @@
 
 open Core
 module LSP = Lsp_base
+(*
+exception NoPosition of string *)
+
+module type RangeMap = sig
```

Comment:
Thanks to add comments and reorder declarations so that to have, whenever possible, type + constructors + destructors. For instance:
```
  type point (** comment *)
  val create_point : int -> int -> point
  val line : point -> int
  val column : point -> int

  type interval (** ... *)
```

Reply:
Idem in the implementation below.

Reply:
Why not moving RangeMap and M into src/core/pos.ml, and rename them as follows?
```
module type RangeMapSig = ...
module RangeMap : RangeMapSig = ...
```

Reply:
I'd suggest going even further and placing the rangemap in its own utility module. Then pos could provide the implementation. I think LP doesn't have a `src/lib` , IMHO OCaml's stdlib is a bit limited so at some LP is gonna need a few extra base modules.

Reply:
They are now in their own utility; they are not perfect but should be pretty easy to tweak them further.